File: DRRR.Server/Dtos/ChatRoomSearchResponseDto.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace DRRR.Server.Dtos
{
    /// <summary>
    /// 聊天室响应DTO
    /// </summary>
    public class ChatRoomSearchResponseDto
    {
        /// <summary>
        /// 聊天室集合
        /// </summary>
        public List<ChatRoomDto> ChatRoomList { get; set; }

        /// <summary>
        /// 分页信息
        /// </summary>
        public PaginationDto Pagination { get; set; }
    }
}

File: DRRR.Client/src/app/chat-rooms/dtos/chat-room-search-response.dto.ts
import { ChatRoomDto } from './chat-room.dto';
import { PaginationDto } from './pagination.dto';

/**
 * 聊天室列表检索响应DTO
 */
export interface ChatRoomSearchResponseDto {
  /**
   * 聊天室列表
   */
  chatRoomList: ChatRoomDto[];

  /**
   * 分页信息
   */
  pagination: PaginationDto;
}
